frontend/src/app/models/base-filter.ts

```diff
@@ -1,67 +1,76 @@
-import { ParamMap } from '@angular/router';
+import { Params } from '@angular/router';
 import * as _ from 'lodash';
-import { BehaviorSubject, Observable, Subject } from 'rxjs';
+import { BehaviorSubject, Observable } from 'rxjs';
 import { map } from 'rxjs/operators';
+import { StoreSync } from '../store/store-sync';
+import { Store } from '../store/types';
+
+export interface FilterState<FilterData> {
+    active: boolean;
+    data: FilterData;
+}
 
 /**
  * Describes shared attributes and properties for filters.
  */
 export interface FilterInterface<FilterData = any> {
-    update: Subject<void>;
-    active: BehaviorSubject<boolean>;
-    data: BehaviorSubject<FilterData>;
+    state$: BehaviorSubject<FilterState<FilterData>>;
+    update: Observable<void>;
     displayName: string;
     description: string;
     filterType: string;
     currentData: FilterData;
     isDefault$: Observable<boolean>;
+    keyInStore: string;
     set: (data: FilterData) => void;
     reset: () => void;
     activate: () => void;
     deactivate: () => void;
     toggle: () => void;
-    setFromParams: (params: ParamMap) => void;
-    toRouteParam: () => Record<string, any>;
+    storeToState: (params: Params) => FilterState<FilterData>;
+    stateToStore: (state: FilterState<FilterData>) => Record<string, any>;
     dataToString: (data: FilterData) => string;
 }
 
+
 /**
  * Abstract filter class
  *
  * Implements much of the logic to handle activity state.
+ *
+ * Filters are a StoreSync model, but have some wackiness because their internal state is
+ * not isomorphic to the stored state. In the UI, we distinguish between the default state
+ * (inactive, default data), and an inactive filter with non-default data. But this
+ * distinction is irrelevant for querying so it's not stored.
```

Comment:
Words like "wacky" and "irrelevant" are not necessarily encouraging in docstrings. Can you rephrase?